database/contracts.go

```diff
@@ -54,3 +59,11 @@ type PgsqlOnConflictConstrainter interface {
 	// PgsqlOnConflictConstraint returns the primary or unique key constraint name of the PostgreSQL table.
 	PgsqlOnConflictConstraint() string
 }
+
+// TxOrDB is just a helper interface that can represent a *[sqlx.Tx] or *[DB] instance.
+type TxOrDB interface {
+	sqlx.ExtContext
+	sqlx.PreparerContext
+
+	PrepareNamedContext(ctx context.Context, query string) (*sqlx.NamedStmt, error)
+}
```

Comment:
Also the whole thing seems just a subset of DB&Tx. I personally would unexport and move it...

Reply:
Wouldn't you also want to use this interface type elsewhere where a `DB` or `Tx` is required?

Reply:
For me the last insert ID makes only sense for immediate subsequent processing – which just screams _transaction_. So I'd just allow Tx as parameter for now unless/until someone finds an actual use case with DB. (But then I'd:

1. unexport the interface OR
2. aggregate there all interfaces and non-interface methods both Tx and DB implement.)